server/src/main/java/org/elasticsearch/snapshots/SnapshotUtils.java

```diff
@@ -118,4 +125,44 @@
         }
         return List.copyOf(result);
     }
+
+    /**
+     * Finds the {@link IndexId}s in the given source snapshot that match the given index patterns for a snapshot clone operation.
+     *
+     * @param sourceSnapshotId source snapshot id
+     * @param targetSnapshot   target snapshot that the clone operation would create
+     * @param repositoryData   repository data of the repository that the clone
+     * @param indexPatterns    index patterns to clone from source- to target snapshot
+     * @return list of index ids to clone
+     * @throws SnapshotException on failure to find concrete request index ids or any index ids
+     */
+    static List<IndexId> findIndexIdsToClone(SnapshotId sourceSnapshotId, Snapshot targetSnapshot, RepositoryData repositoryData,
```

Comment:
Can't we just reuse or adapt `filterIndices()` so that restore and clone indices parameter work the same?

Reply:
:facepalm: yea let's do that and use `IndicesOptions` on the request, that makes it all fall into place much more nicely :) Thanks!